src/Kdyby/Doctrine/DI/EntityLocator.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * This file is part of the Kdyby (http://www.kdyby.org)
+ *
+ * Copyright (c) 2008 Filip Procházka (filip@prochazka.su)
+ * Copyright (c) 2016 Jaroslav Hranička (hranicka@outlook.com)
+ *
+ * For the full copyright and license information, please view the file license.txt that was distributed with this source code.
+ */
+
+namespace Kdyby\Doctrine\DI;
+
+use Doctrine;
+use Kdyby;
+use Nette;
+use Nette\DI\Container;
+
+class EntityLocator
+{
+
+	const CACHE_NS = 'Doctrine.EntityLocator';
+
+	/** @var Nette\Caching\Cache */
+	private $cache;
+
+	/** @var array */
+	private $map = [];
+
+	public function __construct(Nette\Caching\IStorage $storage)
+	{
+		$this->cache = new Nette\Caching\Cache($storage, self::CACHE_NS);
+	}
+
+	public function setup(Container $container, string $containerFile, array $queue, array $classNames, array $managers)
+	{
+		if (empty($queue)) {
+			return;
+		}
+
+		$this->map = $this->cache->load($containerFile);
+
+		if ($this->map === NULL) {
+			foreach ($queue as $manager => $items) {
```

Comment:
This is extracted code from OrmExtension with minor changes (`$config = $this->managerConfigs[$manager];` -> `$repository = $classNames[$manager];`).